src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1288,6 +1369,13 @@ ss::future<add_paritions_tx_reply> tx_gateway_frontend::do_add_partition_to_tx(
     std::vector<begin_tx_reply> brs;
     auto retries = _metadata_dissemination_retries;
     auto delay_ms = _metadata_dissemination_retry_delay_ms;
+    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(
+      kafka::group_id(tx.id));
+    vassert(
+      tx_ntp_opt,
+      "Topic {} doesn't exist in metadata cache",
+      model::tx_manager_nt);
+
```

Comment:
don't make the same decision multiple times - pass partition id as a argument to do_add_partition_to_tx